docs/src/man/basics.md

```diff
@@ -1510,32 +1510,96 @@ julia> german[Not(5), r"S"]
 
 ## Basic Usage of Transformation Functions
 
-In DataFrames.jl we have five functions that we can be used to perform transformations of columns
-of a data frame:
-
-- `combine`: creates a new data frame populated with columns that are results of transformation
-  applied to the source data frame columns, potentially combining its rows;
-- `select`: creates a new data frame that has the same number of rows as the source data frame
-  populated with columns that are results of transformations applied to the source data frame
-  columns;
-- `select!`: the same as `select` but updates the passed data frame in place;
-- `transform`: the same as `select` but keeps the columns that were already present in the data frame
-  (note though that these columns can be potentially modified by the transformation passed to `transform`);
-- `transform!`: the same as `transform` but updates the passed data frame in place.
-
-The fundamental ways to specify a transformation are:
-
-- `source_column => transformation => target_column_name`;
-  In this scenario the `source_column` is passed as an argument to `transformation` function
-  and stored in `target_column_name` column.
-- `source_column => transformation`;
-  In this scenario we apply the transformation function to `source_column` and the
-  target column names is automatically generated.
-- `source_column => target_column_name` renames the `source_column` to `target_column_name`.
-- `source_column` just keep the source column as is in the result without any transformation;
-
-These rules are typically called transformation mini-language.
-
+In DataFrames.jl there are seven functions that can be used to transform data frame columns:
+
+| Function     | Memory Usage                     | Column Retention                             | Row Retention                                       |
+| ------------ | -------------------------------- | -------------------------------------------- | --------------------------------------------------- |
+| `transform`  | Creates a new data frame.        | Retains both source and transformed columns. | Retains same number of rows as source data frame.   |
```

Comment:
maybe there is a better word than *transformed*? the reason is that we have `transform` function. I think it would be clearest if this "word" would be first introduced like:

> all the functions take a data frame as a first argument and then a list of operations that should be performed on the columns of this data frame

In this wording we could later use *operation* word.

(I am giving vague suggestions as I am not a native speaker and here every word matters)

Reply:
`operation` is good.

Yes, I'm not totally happy with having `transform`, `transformation_function`, and `transformation` all mean different things.

Similarly, I somewhat dislike the name `function` in `source_column_selector => function` since you cannot write:
```julia
source_column_selector = "x"
function(x) = 2x #error
transform(df, source_column_selector => function) #error
```
However, I ultimately left the `function` name since I think it is the most descriptive.

Reply:
So let us agree on the following terminology:
```
source_column_selector => operation_function => new_column_names
```
and the whole `Pair` syntax is `operation`. The interpretation of `operation` is defined in `operation mini-language`. The functions that interpret `operation` using the `operation mini-language` meaning are called `functions performing operations` or `functions supporting operation mini-language` (both forms are acceptable).

If we (@nathanrboyer, @nalimilan and @bkamins + @pdeffebach if available) agree on this terminology (i.e. if in this thread we all agree that this is what we like) then:
* this PR should be accordingly adjusted (both in "code" sections and in descriptions)
* whole documentation and manual should be re-written to use it consistently (in a separate PR; I can do this if no-one else wants after this PR is merged)

@nathanrboyer - before making these changes let us first agree that we like the wording. Let me summarize it in bullets again:

1. `source_column_selector => operation_function => new_column_names`
2. whole `Pair` syntax is `operation`.
3. interpretation of `operation` is defined in `operation mini-language`
4. functions (8 of them in total) that interpret `operation` using the `operation mini-language` meaning are called `functions performing operations` or `functions supporting operation mini-language` (both forms are acceptable).

Reply:
3. I prefer `operation syntax`. The language is still Julia. 🤷‍♂️
4. I count 7 functions in my table. What am I missing?
    Could these be called `manipulations` or `manipulation functions`?
    I think it would have to clarified to be "functions performing DataFrame.jl `operations`" since all functions perform operations in general.

Reply:
1. `operation syntax` is OK
2. yes `7` (I just multiplied 2x4, but forgotten that `combine` does not have `!` variant :))
3. `manipulation` instead of `operation` is OK for me (could you please just translate my four bullets under your proposal so that it is clear what you think would be clearest for you). Thank you!

Reply:
In my post above I was suggesting the following:
1. `source_column_selector => operation_function => new_column_names`
2. whole Pair syntax is `operation`
3. interpretation of `operation` is defined in `operation syntax`
4. functions that interpret `operation` using the `operation syntax` are called `data frame manipulation functions`

Reply:
I am OK with this. @nalimilan + @pdeffebach : agreed?

Reply:
I'll also suggest:

1. `source_column_selector => manipulation_function => new_column_names`
2. the whole `Pair` syntax is a `manipulation`
3. interpretation of `manipulations` is `manipulation syntax`
4. functions that interpret `manipulation syntax` are `functions supporting manipulation syntax` or `functions for manipulating dataframes`

Reply:
Sorry, I had missed this thread. I'm fine with the "operation" terminology from https://github.com/JuliaData/DataFrames.jl/pull/2907#discussion_r793008180, which I think we already use in a handful of places (though most of the time we indeed say "transformation"). Though note that this will require adapting docstrings to be fully consistent (should be easy).

I'm less fond of saying "manipulation", as to me this refers to functions such as `combine`, `select` and `transform`, rather than to the *operations* that are passed to them using the pair syntax.

Reply:
> 4. Functions that interpret `operation`s using `operation syntax` are called `data frame manipulation functions`.

@nalimilan, do you want to propose anything different for the above? `functions performing operations` from @bkamins suggestion felt too general to be referring only to the these seven functions. Can you think of a way to make this more clear?

Reply:
Okay, I just wanted to make sure you were okay with "manipulation" in that context.